sprocket-doc.md

```diff
@@ -0,0 +1,77 @@
+
+## Custom themes
+
+While it is technically possible to supply your own custom CSS styling, this capability is currently undocumented. We recommend you stick with the default styling at this point in time, but do let us know what kinds of customization you would like to see in future releases! 
+
+## Per WDL file `index.html`
+
+At a minimum, the `index.html` associated with each WDL file will have a table of contents with links to each struct, task, or workflow documentation page in that directory. If the WDL file has a "preamble" (a comment block before the version statement where each line starts with `##`), that will be rendered as Markdown text above the table of contents.
+
+Example preamble:
+
+```wdl
+## # This is a header
+##
+## This is a paragraph with **bolding**, _italics_, and `code` formatting
+
+version 1.2
+
+workflow foo {}
+```
+
+## v1.0 and v1.1 structs
+
+The WDL specification does not offer a way to document structs prior to WDL v1.2, so the pages for them are rather limited. The pages for these structs will have a copy of the raw WDL definition.
+
+## v1.2 structs
+
+Structs defined in v1.2 WDL _will_ eventually get rich HTML documentation similar to tasks and workflows. However this has not yet been implemented, so they are given the same treatment as pre-v1.2 structs.
+
+## Parameters (inputs and outputs)
+
+Each input and output to a workflow or task should be documented, but there is some flexibility in the specifics.
+
+### Inputs
+
+Each entry in the `input` section of a task or workflow is expected to have a corresponding entry in the `parameter_meta` section. To get the most out of `sprocket doc`, it's recommended that each input entry be an object. That object should have at least a `description` key. There is special handling for the following sub-keys:
+
+- `group`: all inputs sharing the same `String` value will be rendered together in a dedicated table
+    - required inputs are _always_ rendered under the "Required Inputs" table and thus should _not_ have a `group` key (it will be ignored if present)
+    - The `Common` group of inputs will always come after the required inputs
+    - Inputs without a `group` will be rendered under "Other Inputs" which will be the last input table
+    - The `Resource` group of inputs will immediately precede the "Other Inputs" table
+    - All other groups will render alphabetically between the `Common` table and the `Resource` table.
+- `help`: will render at the top of the "Additional Meta" cell of the table
+
+If an input has a `String` value for its parameter meta entry instead of a meta object, that string value will be treated as if it were the `description` key of a meta object.
+
+### Outputs
+
+Outputs should be documented in one of two places: either in the task/workflow `meta` section under an `outputs` key or at the root of the `parameter_meta` section.
```

Comment:
A how-to vs. reference moment: if the former, we should recommend one or the other; if the latter, we should describe differences between the two choices and what happens if both are present.